debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -248,29 +251,15 @@ else if (currentState == State.RESTARTING) {
                     LOGGER.info("Attempting to restart task.");
                     this.coordinator = start(config);
                     LOGGER.info("Successfully restarted task");
+                    restartDelay = null;
+                    setTaskState(State.RUNNING);
                     result = true;
                 }
                 else {
                     LOGGER.info("Awaiting end of restart backoff period after a retriable error");
-                    result = false;
+                    Metronome.parker(Duration.of(retriableRestartWait.toMillis(), ChronoUnit.MILLIS), Clock.SYSTEM).pause();
```

Comment:
Aren't you converting `Duration` to `Duration` here?

Reply:
ops, yes:-( Fixed.